Makefile

```diff
@@ -1,2 +1,151 @@
-all:
-	gcc src/main.c src/user_manual.c src/is_reach_goal.c src/win_game.c src/print_menu.c src/credit.c src/single_player.c src/game_over.c src/take_heart.c src/is_player_hit.c src/is_barrier_hit.c src/load_map.c src/update_player_pos.c src/generate_view.c src/update_barrier.c src/update_arrow.c -l sdl2 -l sdl2_ttf -l sdl2_image
+CC=gcc 
+
+LIBFLAGS= -lmingw32 -lSDL2main -lSDL2
+
+#CREATE BIN AND BUILD FOLDERS TO SAVE THE COMPILED FILES DURING RUNTIME IF THEY DO NOT EXIST
+bin_folder := $(shell mkdir -p bin) 
+build_folder := $(shell mkdir -p build) 
+results_folder := $(shell mkdir -p program_output) 
+
+#TARGET TO GENERATE THE OBJECT FILES 
+main.o: src/main.c
+	$(CC) -g -c $(LIBFLAGS) src/main.c -o build/main.o
+	
+credit.o: src/credit.c
+	$(CC) -g -c $(LIBFLAGS) src/credit.c -o build/credit.o
+	
+game_over.o: src/game_over.c
+	$(CC) -g -c $(LIBFLAGS) src/game_over.c -o build/game_over.o
+	
+generate_view.o: src/generate_view.c
+	$(CC) -g -c $(LIBFLAGS) src/generate_view.c -o build/generate_view.o
+	
+is_barrier_hit.o: src/is_barrier_hit.c
+	$(CC) -g -c $(LIBFLAGS) src/is_barrier_hit.c -o build/is_barrier_hit.o
+
+is_player_hit.o: src/is_player_hit.c
+	$(CC) -g -c $(LIBFLAGS) src/is_player_hit.c -o build/is_barrier_hit.o
+
+is_reach_goal.o: src/is_reach_goal.c
+	$(CC) -g -c $(LIBFLAGS) src/is_reach_goal.c -o build/is_reach_goal.o
+
+load_map.o: src/load_map.c
+	$(CC) -g -c $(LIBFLAGS) src/load_map.c -o build/load_map.o
+
+print_menu.o: src/print_menu.c
+	$(CC) -g -c $(LIBFLAGS) src/print_menu.c -o build/print_menu.o
+
+single_player.o: src/single_player.c
+	$(CC) -g -c $(LIBFLAGS) src/single_player.c -o build/single_player.o
+
+take_heart.o: src/take_heart.c
+	$(CC) -g -c $(LIBFLAGS) src/take_heart.c -o build/take_heart.o
+
+update_arrow.o: src/update_arrow.c
+	$(CC) -g -c $(LIBFLAGS) src/update_arrow.c -o build/update_arrow.o
+
+update_barrier.o: src/update_barrier.c
+	$(CC) -g -c $(LIBFLAGS) src/update_barrier.c -o build/update_barrier.o
+
+update_player_pos.o: src/update_player_pos.c
+	$(CC) -g -c $(LIBFLAGS) src/update_player_pos.c -o build/update_player_pos.o
+	
+user_manual.o: src/user_manual.c
+	$(CC) -g -c $(LIBFLAGS) src/user_manual.c -o build/user_manual.o
+	
+win_game.o: src/win_game.c
+	$(CC) -g -c $(LIBFLAGS) src/win_game.c -o build/win_game.o
+	
+	
+choose_menu_item_test.o: test/choose_menu_item_test.c
```

Comment:
Do the same for test files as well. Add includes and  change the ".o" files.
example:
bulid\choose_menu_item_test.o: test/choose_menu_item_test.c $(INCLUDES)